src/plugins/intel_cpu/src/nodes/scaled_attn.cpp

```diff
@@ -0,0 +1,806 @@
+        k_input.reset(node->getParentEdgeAt(1)->getMemoryPtr());
+        v_input.reset(node->getParentEdgeAt(2)->getMemoryPtr());
+        if (input_num > 3) {
+            // attn_mask
+            if (node->getOriginalInputPrecisionAtPort(3) == Precision::U8) {
+                // bool->f32
+                prepare_attn_mask(node);
+            } else {
+                attn_mask.reset(node->getParentEdgeAt(3)->getMemoryPtr());
+            }
+        }
+        if (input_num > 4) {
+            scale_input = *reinterpret_cast<float*>(node->getParentEdgeAt(4)->getMemoryPtr()->getData());
+        }
+
+        size_t B, H, L1, L0, S;
+
+        // q, k, v: [B, H, L1, S]
+        B = q_input.size(0);
+        H = q_input.size(1);
+        L1 = q_input.size(2);
+        L0 = k_input.size(2) - L1;
+        S = q_input.size(-1);
+
+        if (has_out_transpose)
+            node->redefineOutputMemory({{B, L1, H * S}});
+        else
+            node->redefineOutputMemory({{B, H, L1, S}});
+
+        ov::intel_cpu::PlainTensor<T> output_emb(node->getChildEdgeAt(0)->getMemoryPtr());
+        PlainTensor<T> present_key, present_value;
+
+        q_input.assert_dims({B, H, L1, S});
+        k_input.assert_dims({B, H, L0 + L1, S});
+        v_input.assert_dims({B, H, L0 + L1, S});
+        m_query_emb = q_input;
+        present_key = k_input;
+        present_value = v_input;
+
+        // if fuse_causal_attn == true:
+        //   attn_mask will not be nullptr
+        //   attn_mask will be [B, 1, L1, L0 + L1]
+        //   causal_mask will be got automatically
+        // else:
+        //   if is_causal:
+        //      attn_mask should be nullptr
+        //      causal_mask will be got automatically
+        //   else:
+        //      attn_mask will not be nullptr
+        //      attn_mask will be [B, 1, 1, L0 + L1]
+        //      causal_mask will be nullptr
+        bool auto_causal;
+        bool use_attn_mask;
+        if (fuse_causal_attn) {
+            assert(attn_mask);
+            attn_mask.assert_dims({B, 1, 1, L0 + L1});
+            auto_causal = true;
+            use_attn_mask = true;
+        } else {
+            if (is_causal) {
+                auto_causal = true;
+                use_attn_mask = false;
+            } else {
+                // no attn_mask but has scale, there is a 1-d fake attn_mask
+                if (input_num > 3 && attn_mask.m_rank > 1) {
+                    assert(attn_mask);
+                    auto num = std::accumulate(attn_mask.m_dims, attn_mask.m_dims + attn_mask.m_rank, size_t{1}, std::multiplies<size_t>());
+                    num /= B * (L0 + L1);
+                    attn_mask = attn_mask.reshape({B, 1, num, L0 + L1});
+                    auto_causal = false;
+                    use_attn_mask = true;
+                } else {
+                    auto_causal = false;
+                    use_attn_mask = false;
+                }
+            }
+        }
+
+        if (L1 > 1) {
```

Comment:
Maybe we can add some comment here on why single token requires a special logic, what's the benefit

Reply:
Updated.